### PR TITLE
set up immutable attribute for obsoleted resources

### PR DESCRIPTION
**Pull request by hyi**
@pkdash Could you give a quick review and see if you can +1? Only a couple of line changes, so should be a very quick review.

### CODE REVIEW

**Comment by hydrobuild** [comment]
Can one of the admins verify this patch?

File: hs_core/hydroshare/resource.py [hunk 1]
```diff
@@ -580,6 +580,8 @@ def create_new_version_resource(ori_res, new_res, user):
 
     # since an isReplaceBy relation element is added to original resource, needs to call resource_modified() for original resource
     utils.resource_modified(ori_res, user)
+    # if everything goes well up to this point, set original resource to be immutable so that obsoleted resources cannot be modified from REST API
+    ori_res.raccess.immutable = True
```
Comment (pkdash) [inline hs_core/hydroshare/resource.py]: You need to save the changes to the raccess object

File: hs_core/hydroshare/resource.py [hunk 2]
```diff
@@ -807,7 +809,8 @@ def delete_resource(pk):
         if obsolete_res.metadata.relations.all().filter(type='isReplacedBy').exists():
             eid = obsolete_res.metadata.relations.all().filter(type='isReplacedBy').first().id
             obsolete_res.metadata.delete_element('relation', eid)
-
+            # also make this obsoleted resource editable now that it becomes the latest version
+            obsolete_res.raccess.immutable = False
```
Comment (pkdash) [inline hs_core/hydroshare/resource.py]: Same here

**Comment by hyi** [comment]
@pkdash Thanks for catching that. See if it looks good now.

**Comment by pkdash** [comment]
+1

**Pull request merged**